doc/requirements/Requirements.md

```diff
@@ -0,0 +1,704 @@
+<why>The reference implementation uses a boolean expression based engine.</why>
+
+## Selection criterion
+
+A selection criterion **MUST** have one, and only one, state at a given time.
+
+A selection criterion **MUST** know at construction all possible states.
+<why>To be able to validate: -- rules on start -- state changes</why>
+
+The selection criterion possible states **MUST** be specifiable by directly a
+state set (`Input -> states == identity`)
+<nb>called **exclusive criterion**</nb>
+<nb>An empty set is not allowed as the criterion could not have a state.</nb>
+<why>any criterion can be crated with this one.</why>
+
+The selection criterion possible states **SHOULD** be specifiable by a combination
+of values
+<nb>combination in the mathematical sense `"ab" -> ["", "a", "b", "ab"]`</nb>
+<nb>called **inclusive criterion**</nb>
+<nb>An empty value set is allowed as its combination -- a set containing the
+empty set -- would not be empty. The empty set would be the only possible
+criteria state.</nb>
+<why>The reference implementation supports it.</why>
+
+The PF **MUST NOT** limit the number of criteria.
+
+The PF **MUST NOT** limit the number of states of any given criterion
+<ko>The reference implementation only supports 32 values for an inclusive
+criterion and 2^32 values for an exclusive criterion</ko>
+
+### Definitions
+<dl>
+<dt>Selection criterion rule</dt>
+<dd>
+Function (in the mathematical sense) that **MUST** given selection criteria
+return a Boolean.
+</dd>
+
+<dt>Rule</dt>
+<dd>
+A Boolean expression of Selection criterion rules.
+<nb>implementation only allows AND and OR combination</nb>
+<dd>
+</dl>
+
+### Criterion change
+
+The API to change criterion values **MUST** allow atomicity regarding
+configuration application. I.e. it **MUST** be possible to change multiple
+criterion values without triggering a configuration application.
+<why>Two criterion might have an excluding state. If configuration application
+was triggered after each</why>
+
+### Rules
+
+It **MUST** always be able to express a selection criterion rule from a given
+selection criterion
+I.e.: a criteria **MUST** always have a state that can be matched by a rule.
+<why>if no rules can be formulated from a criterion, it is useless</why>
+
+Parameter values change **SHOULD** be selected by Rules.
+<why>A rule based inference engine has been chosen based on implementation and
+configuration ease</why>
+
+It **MUST** be possible to express a Rule that is always True.
+<why>In order to make a configuration "always applicable"</why>
+
+## Domains
+
+### Definition
+<dl>
+<dt>Domain</dt>
+<dd>
+Ordered set of configuration, all of which contain the values for the
+same parameters.
+</dd>
+</dl>
+
+### Requirement
+
+Each configuration **SHOULD** be in a "domain" (see Definition chapter).
+<why>Domains are mostly a way to define the priority of "Scenarios" for some
+parameters. It is not a MUST because this goal could also be achieve with (for
```

Comment:
Given the why if this req, I must agree. The requirement is about configuration priority. Will reformulate. :-1: